### PR TITLE
Remove usage of the SHELL module or get rid of pipe usage

### PR DESCRIPTION
**Pull request by redhatrises**
- Removes use of SHELL module where it is not really needed
- Removes use of SHELL+PIPE combination where it may not be needed as well

### CODE REVIEW

File: linux_os/guide/system/software/integrity/software-integrity/rpm_verification/rpm_verify_hashes/ansible/shared.yml [hunk 1]
```diff
@@ -14,21 +14,30 @@
   when: (ansible_distribution == "RedHat" or ansible_distribution == "OracleLinux")
 
 - name: "Read files with incorrect hash"
-  shell: |
-    set -o pipefail
-    rpm -Va | grep -E '^..5.* /(bin|sbin|lib|lib64|usr)/' | awk '{print $NF}'
+  command: rpm -Va --nodeps --nosize --nomtime --nordev --nocaps --nolinkto --nouser --nogroup --nomode --noconfig --noghost
```
Comment (redhatrises) [inline linux_os/guide/system/software/integrity/software-integrity/rpm_verification/rpm_verify_hashes/ansible/shared.yml]: @shawndwells what are your thoughts on this?

File: linux_os/guide/system/software/integrity/software-integrity/rpm_verification/rpm_verify_ownership/ansible/shared.yml [hunk 2]
```diff
@@ -4,27 +4,24 @@
 # complexity = high
 # disruption = medium
 - name: "Read list of files with incorrect ownership"
-  shell: |
-    set -o pipefail
-    rpm -Va --nofiledigest | awk '{ if (substr($0,6,1)=="U" || substr($0,7,1)=="G") print $NF }'
+  command: rpm -Va --nodeps --nosignature --nofiledigest --nosize --nomtime --nordev --nocaps --nolinkto --nomode
```
Comment (redhatrises) [inline linux_os/guide/system/software/integrity/software-integrity/rpm_verification/rpm_verify_ownership/ansible/shared.yml]: @shawndwells what are your thoughts on this?

File: linux_os/guide/system/software/integrity/software-integrity/rpm_verification/rpm_verify_permissions/ansible/shared.yml [hunk 3]
```diff
@@ -4,20 +4,25 @@
 # complexity = high
 # disruption = medium
 - name: "Read list of files with incorrect permissions"
-  shell: |
-    set -o pipefail
-    rpm -Va --nofiledigest | awk '{ if (substr($0,2,1)=="M") print $NF }'
+  command: rpm -Va --nodeps --nosignature --nofiledigest --nosize --nomtime --nordev --nocaps --nolinkto --nouser --nogroup
```
Comment (redhatrises) [inline linux_os/guide/system/software/integrity/software-integrity/rpm_verification/rpm_verify_permissions/ansible/shared.yml]: @shawndwells what are your thoughts on this?

File: linux_os/guide/system/software/integrity/software-integrity/rpm_verification/rpm_verify_ownership/ansible/shared.yml [hunk 4]
```diff
@@ -4,27 +4,24 @@
...
     warn: False # Ignore ANSIBLE0006, we can't fetch packages with files with incorrect ownership using rpm module
-  with_items: "{{ files_with_incorrect_ownership.stdout_lines | unique }}"
+  with_items: "{{ files_with_incorrect_ownership.stdout_lines | map('regex_findall', '^[.]+[U|G]+.* (\\/.*)', '\\1') | map('join') | select('match', '(\\/.*)') | list | unique }}"
   register: list_of_packages
+  changed_when: False
   when: (files_with_incorrect_ownership.stdout_lines | length > 0)
 
 - name: "Correct file ownership with RPM"
-  command: "rpm --quiet --setugids '{{ item }}'"
+  command: "rpm --setperms '{{ item }}'"
```
Comment (shawndwells) [inline linux_os/guide/system/software/integrity/software-integrity/rpm_verification/rpm_verify_ownership/ansible/shared.yml]: From the ``name``, this should be resetting ownership and not permissions. ``--setugids`` would be appropriate for that according to the manpage at https://linux.die.net/man/8/rpm.

File: linux_os/guide/system/software/integrity/software-integrity/rpm_verification/rpm_verify_permissions/ansible/shared.yml [hunk 5]
```diff
@@ -4,20 +4,25 @@
...
+  args:
+    warn: False # Ignore ANSIBLE0006, we can't fetch packages with files with incorrect permissions using rpm module
+  with_items: "{{ files_with_incorrect_permissions.stdout_lines | map('regex_findall', '^[.]+[M]+.* (\\/.*)', '\\1') | map('join') | select('match', '(\\/.*)') | list | unique }}"
+  register: list_of_packages
+  changed_when: False
+  when: (files_with_incorrect_permissions.stdout_lines | length > 0)
 
 - name: "Correct file permissions with RPM"
-  shell: "rpm --setperms $(rpm -qf '{{ item }}')"
+  command: "rpm --setperms '{{ item }}'"
```
Comment (shawndwells) [inline linux_os/guide/system/software/integrity/software-integrity/rpm_verification/rpm_verify_permissions/ansible/shared.yml]: Good find/bug fix.

File: shared/templates/template_ANSIBLE_mount_option_remote_filesystems [hunk 6]
```diff
@@ -5,19 +5,16 @@
 # disruption = medium
 
 - name: "Get nfs and nfs4 mount points, that don't have {{{ MOUNTOPTION }}}"
-  shell: |
-    set -o pipefail
-    grep -E "[[:space:]]nfs[4]?[[:space:]]" /etc/fstab | grep -v "{{{ MOUNTOPTION }}}" | awk '{print $2}'
-  args:
-    executable: /bin/bash
+  command: findmnt --fstab --types nfs,nfs4 -O no{{{ MOUNTOPTION }}} -n
```
[the remark this answers is not in the document]
Comment (redhatrises) [reply shared/templates/template_ANSIBLE_mount_option_remote_filesystems]: @shawndwells actually adding the space breaks the command here. Per the manpage, you have to had `no` in front of the option e.g. `nosec`.

File: linux_os/guide/system/software/integrity/software-integrity/rpm_verification/rpm_verify_ownership/ansible/shared.yml (hunk 4 above)
Comment (redhatrises) [reply linux_os/guide/system/software/integrity/software-integrity/rpm_verification/rpm_verify_ownership/ansible/shared.yml]: fixed

**Comment by redhatrises** [comment]
@shawndwells thanks for comments... This is good to go for review.

File: linux_os/guide/system/accounts/accounts-session/root_paths/accounts_root_path_dirs_no_write/ansible/shared.yml [hunk 7]
```diff
@@ -3,27 +3,27 @@
 # strategy = restrict
 # complexity = low
 # disruption = medium
-- name: "Fail if user is not root"
+- name: "Print error message if user is not root"
   fail:
     msg: 'Root account required to read root $PATH'
   when: ansible_user != "root"
+  ignore_errors: true
```
Comment (shawndwells) [inline linux_os/guide/system/accounts/accounts-session/root_paths/accounts_root_path_dirs_no_write/ansible/shared.yml]: Leaving a comment for future documentation:  Idea is for playbook not to fail entirely, rather start skipping remediations if/when needed and continue hardening.

**Pull request merged**